pkg/actions/lua/lakefs/catalogexport/unity_exporter.lua

```diff
@@ -0,0 +1,50 @@
+--[[ TABLE SPECIFICATION:   _lakefs_tables/<table path>
+name: <table name>
+type: delta
+catalog: <catalog name>
+]]
+
+local pathlib = require("path")
+local lakefs = require("lakefs")
+local extractor = require("lakefs/catalogexport/table_extractor")
+--[[
+    - table_descriptors_path: the path under which the table descriptors reside (e.g. "_lakefs_tables").
+      It's necessary that every <table path> in the provided `table_paths` will have a complementary
+      `<table_descriptors_path>/<table path>.yaml` file describing the used Delta Table.
+    - table_paths: a mapping of table name to its location in the object storage
+        { <table name>: <physical location in the object storage> }
+    - databricks_client: a client to interact with databricks.
+    - warehouse_id: Databricks warehouse ID
+
+    Returns a "<table name>: status" map for registration of provided tables.
+]]
+local function register_tables(action, table_descriptors_path, table_paths, databricks_client, warehouse_id)
+    local repo = action.repository_id
+    local commit_id = action.commit_id
+    local branch_id = action.branch_id
+    local response = {}
+    for logical_path, physical_path in pairs(table_paths) do
+        local table_src_path = pathlib.join("/", table_descriptors_path, logical_path .. ".yaml")
+        local table_descriptor = extractor.get_table_descriptor(lakefs, repo, commit_id, table_src_path)
+        if table_descriptor.type ~= "delta" then
+            error("unity exporter supports only table descriptors of type 'delta'. registration failed for table " .. logical_path)
+        end
+        local catalog = table_descriptor.catalog
+        if not catalog then
+            error("catalog name is required to proceed with unity catalog export")
+        end
+        local table_name = table_descriptor.name
+        if not table_name then
+            error("table name is required to proceed with unity catalog export")
+        end
+        local schema_name = databricks_client.create_schema(branch_id, catalog)
+        local status = databricks_client.register_external_table(table_name, physical_path, warehouse_id, catalog, schema_name)
```

Comment:
I suggest to add logging after registering a single table like in [this example](https://github.com/treeverse/lakeFS/blob/master/pkg/actions/lua/lakefs/catalogexport/glue_exporter.lua#L136) I found that to be helpful when things break silently and you're not sure why. 

Maybe even after `local schema_name = databricks_client.create_schema(branch_id, catalog)`

Reply:
If an error occurs in either [create_schema](https://github.com/treeverse/lakeFS/pull/7164/files#diff-ae840c9b54364777462da8226c43a56b0289fd121504a331bf4da23ed050b3c3R108) or [register_external_table](https://github.com/treeverse/lakeFS/pull/7164/files#diff-ae840c9b54364777462da8226c43a56b0289fd121504a331bf4da23ed050b3c3R93), it will be printed..

Reply:
100%, that's why I said **when things break silently**, in glue export we had this experience where user tried to understand why the export didn't work, in AWS - missing data, in Action logs - all good, no info.
The issue was the conceptually there was no error so a function returned all the time with no further action (bug). 

So this logs feedback while watching the action logs especially when troubleshooting things is very helpful kind of "checkpoint" to get feedback on progress.
It's not like there's a lot of info and it'll be garbage. 
Also only when opts.debug == true. 

Thoughts?

Reply:
Concerning the glue export example, if AWS can return missing data (with no error) then, as mentioned (bug), the code should validate that it's not the case.
I don't see places where such cases can happen in the code.
Nonetheless, if it breaks silently, I'll fiercely eat my hat, but for now, I don't think we should add extra code for this case (even though it's minimal).

Reply:
🎩